src/components/BooksQuery.tsx

```diff
@@ -1,45 +1,42 @@
-import axios from "axios";
 import { useState } from "react";
 import Card from "@mui/material/Card";
 import "./books.css";
 import { CardContent, Typography } from "@mui/material";
-type book = {
-  id: number;
-  title: string;
-  author: string;
-  year: string;
-};
+import { Book } from "../types/book";
+import { AllBooks } from "../types/allBooks";
+import { ResponseBook, Items } from "../types/respBooks";
+import { GoogleBook } from "./API/GoogleBook";
 const BooksQuery = () => {
   const [book, setBooks] = useState("");
-  const [list, setList] = useState([]);
-  const submit = (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
+  const [list, setList] = useState<AllBooks>({ books: [] });
+  const submit = async (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
     e.preventDefault();
     if (!book) {
       alert("Please enter a book name");
     } else {
-      getData();
+      const data = await getData();
+      setList(setBooksModel(data));
     }
-    console.log(list);
   };
-  const getData = () => {
-    axios
-      .get(`https://www.googleapis.com/books/v1/volumes?q=${book}`)
-      .then((data) => {
-        setList(setBooksModel(data.data.items));
-      });
+  const getData = async () => {
```

Comment:
This `getData` method 

extract this fetching logic into a custom hook for example `useSearchBooks` (we will make it more generic later)

it should be consumed and used like this in your components

```tsx
const [{data, loading, error}, searchBooks] = useSearchBooks();

// used later
searchBooks(searchTerm)
```